### PR TITLE
Switch from black to ruff-format

### PR DESCRIPTION
**Pull request by brisvag**
# References and relevant issues
Re-opening #6383, now that the issues raised by jni https://github.com/napari/napari/pull/6386#issuecomment-1780604381 are resolved by merging #6407.

# Description

Moves away from black in favour of unifying everything under ruff with ruff-format. See 514938c18 for the actual changes to configurations without all the formatting updates.

The formatting updates come from running `pre-commit run --all-files`.

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
Main thing I notice is the change from numpy one-line docstrings (where the terminating triple quote is own line) to pep257 style. No preference here, so I'm fine with this, since we already have ruff anyways.

The manifest fail is due to lack of black:
https://github.com/napari/napari/actions/runs/8438077981/job/23114126982?pr=6775#step:6:63

The typing issue seems due to the line breaks making the `ignore` not apply to the right line?

Ubuntu fail is a hash issue unrelated.

**Review by Czaki — APPROVED** [review]
Looks ok for me

File: pyproject.toml [hunk 1]
```diff
@@ -300,12 +275,14 @@ ignore = [
     "E501", "TCH001", "TCH002", "TCH003",
     "A003", # flake8-builtins - we have class attributes violating these rule
     "COM812", # flake8-commas - we don't like adding comma on single line of arguments
+    "COM819", # conflicts with ruff-format
     "SIM117", # flake8-simplify - we some of merged with statements are not looking great with black, reanble after drop python 3.9
     "RET504", # not fixed yet https://github.com/charliermarsh/ruff/issues/2950
```
Comment (psobolewskiPhD) [inline pyproject.toml:280]: that issue was closed as implemented in a PR, worth revisiting?

File: napari/conftest.py [hunk 2]
```diff
@@ -671,8 +671,8 @@ def _check_throttle_info(path):
             return (
                 path
                 + " it's possible that there was a problem with unfinished work by a "
```
Comment (psobolewskiPhD) [inline napari/conftest.py:673]: why did these get left with double-quote but the next lines were changed?

**Review by psobolewskiPhD — APPROVED** [review]
Left two questions, but everything seems good to me.

File: napari/conftest.py (hunk 2 above)
Comment (Czaki) [reply napari/conftest.py:673]: Because of `it's` that contains single quite in it.

File: pyproject.toml (hunk 1 above)
Comment (Czaki) [reply pyproject.toml:280]: Maybe yes, but in separate PR.

**Comment by Czaki** [comment]
@jni You may have strongest opinion here.

File: pyproject.toml (hunk 1 above)
Comment (brisvag) [reply pyproject.toml:280]: @psobolewskiPhD could you link to that PR?

Comment (psobolewskiPhD) [reply pyproject.toml:280]: https://github.com/astral-sh/ruff/pull/3393
but there were two related followups
https://github.com/astral-sh/ruff/pull/4997
https://github.com/astral-sh/ruff/pull/4998

**Review by jni — APPROVED** [review]

File: napari/utils/action_manager.py [hunk 3]
```diff
@@ -347,9 +347,9 @@ def _get_layer_shortcuts(self, layers) -> dict:
                 action = self._actions.get(name, None)
                 if action and layer == action.keymapprovider:
                     for shortcut in shortcuts:
-                        layer_shortcuts[layer][
-                            str(shortcut)
-                        ] = action.description
+                        layer_shortcuts[layer][str(shortcut)] = (
+                            action.description
+                        )
```
Comment (jni) [inline napari/utils/action_manager.py:352]: Oh sweet mercy, at long last 😂😭

**Pull request merged**